### PR TITLE
S3 to DropBox file upload

### PR DESCRIPTION
**Pull request by jerinpetergeorge**

### CODE REVIEW

File: manage.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
```
Comment (patroqueeet) [inline manage.py:1]: if this is used as a lib, there is no need for a manage.py file. `dbbackup` project structure should be a good place to see it. https://github.com/Develatio/django-walletpass README.txt at the very bottom has a docker cmd to run tests/lib cmd for you

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+django~=4.1.5
```
Comment (patroqueeet) [inline requirements.txt]: use setup.py for deps. see wallet pass app mentioned above for a template

Comment (patroqueeet) [inline requirements.txt]: Django is a requirement of the others, so we should be able to skip here. we are with 3.2 as of now, so please start with that version earliest.

File: s3_dropbox_sync/asgi.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+"""
```
Comment (patroqueeet) [inline s3_dropbox_sync/asgi.py:1]: can be deleted

File: s3_dropbox_sync/settings.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
+"""
```
Comment (patroqueeet) [inline s3_dropbox_sync/settings.py:1]: drop the Django defaults, only lib spec settings needed. learn here: https://github.com/jazzband/django-dbbackup/blob/master/dbbackup/settings.py

File: s3_dropbox_sync/urls.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+"""s3_dropbox_sync URL Configuration
```
Comment (patroqueeet) [inline s3_dropbox_sync/urls.py:1]: no urls required here I assume

File: s3_dropbox_sync/wsgi.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+"""
```
Comment (patroqueeet) [inline s3_dropbox_sync/wsgi.py:1]: drop that unneeded one too pls

File: s3_dropbox_sync/sync.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            tarinfo.size = len(media_file)
+            tar_file.addfile(tarinfo, media_file)
+        # Close the TAR for writing
+        tar_file.close()
+        return file_obj
+
+    def generate_backup(self):
+        return self._create_tar_file_object()
+
+    def write_to_storage(self, file, path):
```
Comment (patroqueeet) [inline s3_dropbox_sync/sync.py]: where is that used?

**Review by patroqueeet — CHANGES_REQUESTED** [review]
questions: 

- is the tarball deleted after backup?
- I'm missing the management command to run it from local projects manage.py
- how did you test, that it works? feel free to quickly drop such information into an README.txt

File: manage.py (hunk 1 above)
Comment (jerinpetergeorge) [reply manage.py:1]: There are many cookiecutter templates available to create PyPackage (see [this one](https://github.com/audreyfeldroy/cookiecutter-pypackage) and also [similar ones](https://github.com/audreyfeldroy/cookiecutter-pypackage#similar-cookiecutter-templates)).

Do you have any preferences, or shall I just start with a simple package directory style?

**Comment by jerinpetergeorge** [comment]
> questions:
> 
> * is the tarball deleted after backup?
> * I'm missing the management command to run it from local projects manage.py
> * how did you test, that it works? feel free to quickly drop such information into an README.txt

____________________________
1. The file is **_not deleted explicitly_** after upload (FYI: We're using `db-backup`'s function to create the file, which is configured using `DBBACKUP_TMP_DIR` variable. The default value is `/tmp` or a similar ). So, what's your take on this? IMO, we should delete it after the uploads.
2. Yes, you're. This PR is only meant to indicate the basic sync functionality is working (ie, create backup of an S3 directory and upload to DropBox) . I hope you watched the video that I've shared, else [you can find the same here](https://app.usebubbles.com/wUz6iM3xX4mkEP1qrvJm9S/demo)
3. Please refer this [video](https://app.usebubbles.com/wUz6iM3xX4mkEP1qrvJm9S/demo)

File: s3_dropbox_sync/sync.py (hunk 7 above)
Comment (jerinpetergeorge) [reply s3_dropbox_sync/sync.py]: Good catch, will be removed.

File: manage.py (hunk 1 above)
Comment (patroqueeet) [reply manage.py:1]: make it simple, as less as possible. no prefs if you want to pick one

**Comment by patroqueeet** [comment]
> > questions:
> > 
> > * is the tarball deleted after backup?
> > * I'm missing the management command to run it from local projects manage.py
> > * how did you test, that it works? feel free to quickly drop such information into an README.txt
> 
> 1. The file is **_not deleted explicitly_** after upload (FYI: We're using `db-backup`'s function to create the file, which is configured using `DBBACKUP_TMP_DIR` variable. The default value is `/tmp` or a similar ). So, what's your take on this? IMO, we should delete it after the uploads.
> 2. Yes, you're. This PR is only meant to indicate the basic sync functionality is working (ie, create backup of an S3 directory and upload to DropBox) . I hope you watched the video that I've shared, else [you can find the same here](https://app.usebubbles.com/wUz6iM3xX4mkEP1qrvJm9S/demo)
> 3. Please refer this [video](https://app.usebubbles.com/wUz6iM3xX4mkEP1qrvJm9S/demo)

1) let's delete
2) k
3) missed the link. will check later

**Pull request merged**